src/y0/graph.py

```diff
@@ -514,6 +524,27 @@ def get_markov_blanket(self, nodes: Union[Variable, Iterable[Variable]]) -> Set[
                 blanket.update(self.directed.predecessors(successor))
         return blanket.difference(nodes)
 
+    def pre(
```

Comment:
@pnavada we need some test cases for this

Reply:
actually I would like to split this into a separate PR since @njmmerrill also has implemented something similar but less generic, then we can reduce some other work elsewhere